identity-account-storage/src/types/signature.rs

```diff
@@ -1,33 +1,20 @@
 // Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use identity_core::crypto::PublicKey;
-#[cfg(feature = "bindings-derive")]
 use serde::Deserialize;
-#[cfg(feature = "bindings-derive")]
 use serde::Serialize;
 
-/// A digital signature and associated public key.
-#[cfg_attr(feature = "bindings-derive", derive(Clone, Deserialize, Serialize))]
-#[cfg_attr(not(feature = "bindings-derive"), derive(Clone))]
-pub struct Signature {
-  pub(crate) pkey: PublicKey,
-  pub(crate) data: Vec<u8>,
+/// A digital signature.
+#[derive(Clone, Deserialize, Serialize)]
+pub enum Signature {
+  Ed25519(Vec<u8>),
```

Comment:
We can, but if we want to add support to something else latter on, we would have to change it.

Reply:
I assume any signature can be represented as bytes, and what type of signature it is, is clear from context, e.g. when calling `key_sign` with a `KeyLocation` that contains a certain `MethodType`. So having to do the additional match seems unnecessary, unless the `Storage` implementation is simply wrong and returns something else. It's maybe slightly safer, but I don't really see a major advantage.
At least judging by `RemoteSign`, I think it's better not to use `SignatureValue`, so we don't accidentally return _that_ signature value, but have to first encode it as base58 and then return an instance of `SignatureValue`.

https://github.com/iotaledger/identity.rs/blob/1271d951cbb6a1ec2940e99d12015b6459a78aaa/identity-account-storage/src/crypto/remote.rs#L53-L56

Reply:
> I assume any signature can be represented as bytes

I'm not sure that's a strong assumption. For instance the [BBS+ Signatures 2020 spec requires a nonce value used to derive the proof](https://w3c-ccg.github.io/ldp-bbs2020/#nonce) as well, so it's probable that we will need more than just the signature/proof bytes in the future. One could argue that the `nonce` could be predetermined and passed to the `sign` function but that just moves the problem to the signing interface not being general enough to handle it.

> At least judging by RemoteSign, I think it's better not to use SignatureValue,

I agree, `SIgnatureValue` does not seem fully appropriate for this as it would require `Storage` implementers to also implement the signature schemes such as `JCSEd25519Signature2020` and handle encoding rather than just the raw key signing.

I'm fine using just `Vec<u8>` or a similar new-type wrapper, until we need to refactor it to handle a new method type, if we cannot come to a consensus on what a future-proof design looks like now.